### PR TITLE
[develop][integ-test] Add GetInstanceProfile permission for integration test permission boundary

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
test_iam_resource_prefix started to fail a few days ago. It seems CloudFormation started to require GetInstanceProfile permission while creating an instance profile. Moreover, the action is ignores the path prefix. Therefore, this commit adds `iam:GetInstanceProfile` and relax the resource specification.

### Tests
test_iam_resource_prefix has been passed

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/iam/test_iam.py [hunk 1]
```diff
@@ -472,15 +472,13 @@ def _create_permission_boundary(permission_boundary_name):
...
                     ],
                     "Effect": "Allow",
                     "Resource": [
                         {"Fn::Sub": "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/${CustomIamPathPrefix}/*"},
                         {"Fn::Sub": "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/${CustomIamNamePrefix}*"},
-                        {
-                            "Fn::Sub": "arn:${AWS::Partition}:iam::${AWS::AccountId}:instance-profile/"
-                            "${CustomIamPathPrefix}/*"
-                        },
+                        {"Fn::Sub": "arn:${AWS::Partition}:iam::${AWS::AccountId}:instance-profile/*"},
```
Comment (himani2411) [inline tests/integration-tests/tests/iam/test_iam.py:481]: why are we relaxing the Resource specification? IIRC, We do add ResourcePrefix to the instance-profile as well

**Review by himani2411 — APPROVED** [review]

File: tests/integration-tests/tests/iam/test_iam.py (hunk 1 above)
Comment (hanwen-pcluste) [reply tests/integration-tests/tests/iam/test_iam.py:481]: GetInstanceProfile requires to relax the Resource spec. I don't understand why. The best we can do is to split the permission to only relax the Resource for GetInstanceProfile, not for other actions.

**Review by gmarciani — APPROVED** [review]
In this PR we are introducing a new action `iam:GetInstanceProfile` that was not required until now and we are relaxing the permission removing the custom prefix from the resource. Let's be sure to understand why this became necessary. Approved because this is a readonly permission and the change only impacts test code.
